src/Models/Link.php

```diff
@@ -446,4 +441,17 @@ public function getDefaultTitle(): string
         }
         return $default;
     }
+
+    /**
+     * This method process the defined singular_name of Link class
+     * to get the short code of the Link class name.
+     * Or If the name is not defined (by redefining $singular_name in the subclass),
+     * this use the class name. The Link prefix is removed from the class name
+     * and the resulting name is converted to lowercase.
+     * Example: Link => link, EmailLink => email, FileLink => file, SiteTreeLink => sitetree
+     */
+    public function getShortCode(): string
+    {
+        return strtolower(str_replace([' ', 'Link'], '', ClassInfo::shortName($this))) ?? '';
```

Comment:
This is a minor tweak.
- `ClassInfo::shortName` won't return spaces, so you don't need to remove spaces.
- It would a bit safer to only remove the `Link` part if it's at the end of the class name. So using `rtrim` would be a bit safer here.



```suggestion
        return strtolower(rtrim(ClassInfo::shortName($this), 'Link')) ?? '';
```

Reply:
Updated